### PR TITLE
Feature/front end scaffold

### PR DESCRIPTION
**Pull request by XanderVertegaal**
**ParsePort front-end**

Features:
- Basic routing (with dummy routes)
- Simple input form with API call for Spindle
- Error handling and alert modal
- Linting (ESLint)

Known issue:
- On small screens (mobile), the hamburger menu overflows. (I have created an issue for this, but we may not intend to support mobile anyway.

### CODE REVIEW

**Comment by XanderVertegaal** [comment]
Tests fail because the runner cannot find `version.ts`. 
image

The file is there in my codebase, but it has apparently been added to `.gitignore` as part of the cookie-cutter set up. Does either of you know what the usual practice is?

**Comment by bbonf** [comment]
> Tests fail because the runner cannot find `version.ts`. image
> 
> The file is there in my codebase, but it has apparently been added to `.gitignore` as part of the cookie-cutter set up. Does either of you know what the usual practice is?

I think this should be generated by `node build/build-pre.js`, but perhaps it's missing from the cookiecutter template? here's an example from another repo: https://github.com/UUDigitalHumanitieslab/gretel-django/blob/develop/web-ui/package.json#L8

**Comment by tijmenbaarda** [comment]
I had the same issue with version.ts not being there. Another problem was that for some reason `frontend/package.json` did not contain the run scripts `stop` and `serve`, so that `yarn start` from the root directory did not work. If I add these things manually the application runs fine!

It seems that both problems occur because there are some things missing in `frontend/package.json`. Perhaps something went wrong in the cookiecutter? In any case you could try copying the commands from the GrETEL repo that Ben pointed at.

**Comment by tijmenbaarda** [comment]
The functional tests are failing because `conftest.py` has the wrong value of `--base-address`. It should be `localhost:4200` instead of `localhost:8000`, because in development mode (after running `yarn start`) the frontend is programmed to proxy backend requests to the server on port 8000. In production mode it is the other way around. Unfortunately this is not well documented and officially `localhost:8000` should work in both development and production mode, but in the current cookiecutter setup it doesn't and that causes some confusion every now and then.

File: frontend/.eslintrc.json [hunk 1]
```diff
@@ -1,50 +1,46 @@
...
+      "files": [
+        "*.html"
+      ],
+      "extends": [
+        "plugin:@angular-eslint/template/recommended"
+      ],
+      "rules": {}
+    }
+  ]
+}
```
Comment (tijmenbaarda) [inline frontend/.eslintrc.json:46]: I see the additions to the `extends` section here, which look good, but what are the other changes?

File: frontend/src/app/shared/services/error-handler.service.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { ErrorHandler, Injectable } from '@angular/core';
+import { AlertService } from './alert.service';
+import { AlertType } from '../components/alert/alert.component';
+import { HttpErrorResponse } from '@angular/common/http';
+
+@Injectable({
+    providedIn: 'root'   
+})
+export class ErrorHandlerService implements ErrorHandler {
```
Comment (tijmenbaarda) [inline frontend/src/app/shared/services/error-handler.service.ts:9]: This is great! But maybe specify that this service is specific for HTTP errors, or make it more generic?

**Review by tijmenbaarda — COMMENTED** [review]
Great! Very nice to have an error and alert service from the beginning. I think it would be good to put the logic that deals with out backend/Spindle API in a separate service, but we can do that later as it gets more complex.

File: frontend/src/app/shared/services/error-handler.service.ts (hunk 2 above)
Comment (XanderVertegaal) [reply frontend/src/app/shared/services/error-handler.service.ts:9]: Yes, good idea! I'll add a class method for generic error messages so we can use it for other errors/notifications as well.

File: frontend/.eslintrc.json (hunk 1 above)
Comment (XanderVertegaal) [reply frontend/.eslintrc.json:46]: I fixed an indenting error, which makes Git think that the whole file was changed. The only other change here is the removal of `parserOptions`. I did not do this deliberately, but I think it was done automatically by adding the `@typescript-eslint/recommended` plugin.

**Comment by XanderVertegaal** [comment]
Tests run successfully now! 🥳  Thanks for the comments and proposed fixes! I have made a few small changes. Have a look if you agree with them; then I will merge this branch into develop.

**Pull request merged**